tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -91,9 +93,20 @@ private bool TryGetGitTagsFromSourceLink([NotNullWhen(true)] out GitMetadata? re
         {
             // Cannot determine the entry assembly. This may mean this method was called too early.
             // Return false to indicate that we should try again later.
-            Log.Debug("Cannot extract SourceLink information as the entry assembly could not be determined.");
-            result = default;
-            return false;
+
+            var nbTries = Interlocked.Increment(ref _tryCount);
+            if (nbTries > 100)
+            {
+                Log.Debug("Tried 100 times to get the SourceLink information. Giving up.");
```

Comment:
Non-blocking comment/question - this ideally should be `Log.Warning`, so that we would get telemetry on it, but since it seems it's expected to happen in unit tests, we'd need to somehow account for that?

Reply:
Indeed. There's a log checker that already ignores some patterns.  We could add this one.

Reply:
I'll address that in a follow-up PR as CI is flakey and I don't want to delay the hotfix release.